soc/xtensa/esp32/linker.ld

```diff
@@ -29,6 +31,10 @@ PROVIDE ( intr_matrix_set = 0x4000681c );
 PROVIDE ( g_ticks_per_us_app = 0x3ffe40f0 );
 PROVIDE ( g_ticks_per_us_pro = 0x3ffe01e0 );
 PROVIDE ( ets_delay_us = 0x40008534 );
+PROVIDE ( gpio_output_set = 0x40009b24 );
```

Comment:
These are part of a block that I think should be documented as to why these have to be global symbols.  Are they all things that will become unnecessary when `esp-idf` is cleaned up?

Reply:
These symbols are used on `hal_espressif` external module branch (i.e. esp-idf). 
To get rid of this "no prefix" symbols, it would require to change the whole esp-idf source as well. In fact, we will change it soon and the majority of the symbols will have `esp_rom` prefix (not `esp32_rom`) in it. Then, this `linker.ld` file won't have these "no prefix" anymore.
I can add a line comment on the beginning of these block, sounds good?
By the way, Is there any guideline regarding linker PROVIDE naming rule?

Reply:
> In fact, we will change it soon and the majority of the symbols will have `esp_rom` prefix (not `esp32_rom`) in it.

That sounds fine.  When that happens update all the Zephyr source with s/esp32_rom/esp_rom/ so everything's consistent.

> I can add a line comment on the beginning of these block, sounds good?

Yes.  Thank you.

> By the way, Is there any guideline regarding linker PROVIDE naming rule?

Probably not.  It's implicit in rules that global symbols should be sufficiently unique to prevent conflict with other Zephyr modules and external libraries that take similar care.  Those rules are probably also unwritten, but should fall under #29569.